sku.0/sys.server/compiled/game/script/gm/cmd.java

```diff
@@ -654,86 +654,149 @@ else if (faction.equalsIgnoreCase("rebel"))
+                command = "increase";
+            } while (currentRank != goalRank);
+        } else if (neededRankChange < 0) {
+            do {
+                gcw.decreaseGcwRatingToPreviousRank(target);
+                currentRank = pvpGetCurrentGcwRank(target);
+                command = "decrease";
+            } while (currentRank != goalRank);
         }
-        else
-        {
-            sendSystemMessageTestingOnly(self, "The system was unable to update (" + target + ") " + getName(target) + "'s rank updated to (" + newrank + ") " + getString(new string_id("faction_recruiter", factions.getRankName(newrank, faction))));
+
+		// it's probably better to read a string file to get these but who has time for that
+		// this is just a rarely used admin script anyways
+        if (faction == -615855020) {
+            if (currentRank == 1) {
+                rankName = "Private";
+            } else if (currentRank == 2) {
+                rankName = "Lance Corporal";
+            }  else if (currentRank == 3) {
+                rankName = "Corporal";
+            } else if (currentRank == 4) {
+                rankName = "Sergeant";
+            } else if (currentRank == 5) {
+                rankName = "Master Sergeant";
+            } else if (currentRank == 6) {
+                rankName = "Sergeant Major";
+            } else if (currentRank == 7) {
+                rankName = "Lieutenant";
+            } else if (currentRank == 8) {
+                rankName = "Captain";
+            } else if (currentRank == 9) {
+                rankName = "Major";
+            } else if (currentRank == 10) {
+                rankName = "Lt. Colonel";
+            } else if (currentRank == 11) {
+                rankName = "Colonel";
+            } else if (currentRank == 12) {
+                rankName = "General";
+            }
+        } else if (faction == 370444368) {
+            if (currentRank == 1) {
+                rankName = "Private";
+            } else if (currentRank == 2) {
+                rankName = "Trooper";
+            }  else if (currentRank == 3) {
+                rankName = "High Trooper";
+            } else if (currentRank == 4) {
+                rankName = "Sergeant";
+            } else if (currentRank == 5) {
+                rankName = "Senior Sergeant";
+            } else if (currentRank == 6) {
+                rankName = "Sergeant Major";
+            } else if (currentRank == 7) {
+                rankName = "Lieutenant";
+            } else if (currentRank == 8) {
+                rankName = "Captain";
+            } else if (currentRank == 9) {
+                rankName = "Major";
+            } else if (currentRank == 10) {
+                rankName = "Commander";
+            } else if (currentRank == 11) {
+                rankName = "Colonel";
+            } else if (currentRank == 12) {
+                rankName = "General";
+            }
+        }
+
+        if (command == "increase") {

```

Comment:
Should probably use .equals() here.

Reply:
Agreed.  .equals() is preferred when comparing strings.